lib/game.rb

```diff
@@ -0,0 +1,145 @@
+
+class Game
+
+  attr_accessor :player_1, :player_2, :board
+
+  WIN_COMBINATIONS =
+  [
+    [0,1,2],
+    [3,4,5],
+    [6,7,8],
+    [0,3,6],
+    [1,4,7],
+    [2,5,8],
+    [2,4,6],
+    [0,4,8]
+            ]
+
+    def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+
+      @player_1 = player_1
+      @player_2 = player_2
+      @board = board
+
+    end
+
+
+    def turn_count
+      counter = 0
+      self.board.cells.map do |current_turn|
+        if current_turn == "X" || current_turn == "O"
+          counter += 1
+        end
+      end
+      counter
+    end
+
+
+    def current_player
+
+      self.turn_count
+      if self.turn_count % 2 == 0
+        self.player_1
+
+      else
+        self.player_2
+
+      end
+    end
+
+
+    def over?
+       self.won? || self.draw?
+
+    end
+
+
+    def won?
+        WIN_COMBINATIONS.any? do |win_combination|
```

Comment:
Here you can use [`Enumerable#detect`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-detect) to return the winning combination or `nil` if none is found.

```ruby
  def won?
    WIN_COMBINATIONS.detect do |combo|
      @board.cells[combo[0]] == @board.cells[combo[1]] &&
      @board.cells[combo[0]] == @board.cells[combo[2]] &&
      @board.taken?(combo[0] + 1)
    end
  end
```